### DISCUSSION TITLE
Stream stdout from Sidecar

### DISCUSSION THREAD

**Discussion by AnupamKris** [tauri-apps/tauri#8641] [Q&A] [answered]
I have a python file converted to exe that runs as a sidecar from Tauri. Is it possible to get the stdout from the sidecar live so that I can display a progressbar for the user. The current way is to use await, but it waits till the execution is complete. I am invoking it from js.

**Comment by FabianLars** [comment]
Use spawn() instead of execute(). The first Command example here https://tauri.app/v1/api/js/shell#command should be what you're asking for.

**Reply by AnupamKris** [reply]
It received the stdouts but only when everything is done.. It gives nothing while executing and when the execution is completed, it flushes the whole output. I used `command.spawn()` with `command.stdout.on("data")`.

**Reply by FabianLars** [reply, marked as answer]
Can you try printing it with `print(message, flush = True)` (instead of just `print(message)`)?

**Reply by AnupamKris** [reply]
Yes, This worked. Thanks..!
